### PR TITLE
fix(linux-client): forbid passing the token as a CLI arg

### LINKED ISSUE #4682: fix(linux-client): forbid / deprecate passing tokens as CLI args
`ps` lets normal users see the CLI args of root users, so they can't be used for secrets

### LINKED ISSUE #4691: chore(linux-client): unset `FIREZONE_TOKEN` env var once we read it, for security

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #4682 
Closes #4691 

```[tasklist]
# Before merging
- [x] Wait for `linux-group` test to go green on `main` (#4692)
- [ ] Wait for those browsers tests to get fixed
- [ ] *All* compatibility tests must pass on this branch
```

### CODE REVIEW

File: rust/headless-client/src/imp_linux.rs [hunk 1]
```diff
@@ -30,14 +30,36 @@ const ROOT_USER: u32 = 0;
...
+    // SAFETY: We haven't spawned any other threads, this code should be the first
+    // thing to run after entering `main`. So nobody else is reading the environment.
+    #[allow(unused_unsafe)]
+    unsafe {
+        // This removes the token from the environment per <https://security.stackexchange.com/a/271285>. We run as root so it may not do anything besides defense-in-depth.
+        std::env::remove_var(TOKEN_ENV_KEY);
+    }
+    assert!(std::env::var(TOKEN_ENV_KEY).is_err());
+
+    let rt = tokio::runtime::Runtime::new()?;
```
Comment (conectado) [inline rust/headless-client/src/imp_linux.rs]: we can go with a single worker thread like we do in the other client applications

File: rust/headless-client/src/lib.rs [hunk 2]
```diff
@@ -70,15 +73,12 @@ struct Cli {
     #[arg(long)]
     check: bool,
 
-    /// Token generated by the portal to authorize websocket connection.
-
-    // TODO: It isn't good for security to pass the token as a CLI arg.
-    // If we pass it as an env var, we should remove it immediately so that
-    // child processes don't inherit it. Reading it from a file is probably safest.
-    #[arg(env = "FIREZONE_TOKEN", hide = true, long)]
```
Comment (conectado) [inline rust/headless-client/src/lib.rs:78]: We can still use `clap` to retrieve the token through the env var, we just need to remove the `long` attribute

File: rust/headless-client/src/lib.rs [hunk 3]
```diff
@@ -71,14 +76,17 @@ struct Cli {
...
     /// Token generated by the portal to authorize websocket connection.
-
-    // TODO: It isn't good for security to pass the token as a CLI arg.
-    // If we pass it as an env var, we should remove it immediately so that
-    // child processes don't inherit it. Reading it from a file is probably safest.
-    #[arg(env = "FIREZONE_TOKEN", hide = true, long)]
-    pub token: Option<String>,
+    // systemd recommends against passing secrets through env vars:
+    // <https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#Environment=>
+    #[arg(env = TOKEN_ENV_KEY, hide = true)]
```
Comment (conectado) [inline rust/headless-client/src/lib.rs:81]: I don't think we need the `hide = true` anymore?

**Review by conectado — APPROVED** [review]
:+1:

File: rust/headless-client/src/lib.rs (hunk 3 above)
Comment (ReactorScram) [reply rust/headless-client/src/lib.rs:81]: I tried removing it and that seemed to make it into a positional argument: `[OPTIONS] [TOKEN] [COMMAND]`

**Pull request merged**